--- Makefile.am.orig 2013-08-06 14:21:53.000000000 +0200
+++ Makefile.am      2013-03-13 13:44:35.000000000 +0100
@@ -16,7 +16,7 @@
   MAKE_DEMO = demo
 endif

-SUBDIRS = json-glib lib pysearpc ${MAKE_DEMO} tests
+SUBDIRS = lib pysearpc ${MAKE_DEMO} tests

 install-data-local:
 if MACOS
--- configure.ac.orig	2013-03-27 11:50:09.000000000 +0100
+++ configure.ac	2013-08-06 14:44:26.000000000 +0200
@@ -89,7 +89,10 @@
    fi
 fi
 
-AC_CONFIG_SUBDIRS([json-glib])
+#AC_CONFIG_SUBDIRS([json-glib])
+PKG_CHECK_MODULES(JSONGLIB, [json-glib-1.0])
+AC_SUBST(JSONGLIB_CFLAGS)
+AC_SUBST(JSONGLIB_LIBS)
 
 AC_CONFIG_FILES([Makefile
                  lib/Makefile
--- lib/Makefile.am.orig	2013-08-06 16:20:01.000000000 +0200
+++ lib/Makefile.am	2013-08-06 16:20:44.000000000 +0200
@@ -1,7 +1,6 @@
 
-AM_CFLAGS = @GLIB_CFLAGS@ \
+AM_CFLAGS = @GLIB_CFLAGS@ @JSONGLIB_CFLAGS@ \
 	-I${top_builddir}/lib \
-	-I${top_srcdir}/json-glib \
 	-I${top_srcdir}/lib \
 	-DG_LOG_DOMAIN=\"Searpc\"
 
@@ -13,7 +12,6 @@
 
 libsearpc_la_LDFLAGS = -version-info 1:2:0  -no-undefined
 
-libsearpc_la_LIBADD = @GLIB_LIBS@ \
-	${top_builddir}/json-glib/json-glib/libsearpc-json-glib.la
+libsearpc_la_LIBADD = @GLIB_LIBS@ @JSONGLIB_LIBS@
 
 dist_bin_SCRIPTS = searpc-codegen.py
