--- CMakeLists.txt.orig	2013-08-06 13:40:05.000000000 +0200
+++ CMakeLists.txt	2013-08-06 13:42:33.000000000 +0200
@@ -29,6 +29,7 @@
 CHECK_INCLUDE_FILES(sys/tree.h HAVE_SYS_TREE)
 CHECK_INCLUDE_FILES(sys/queue.h HAVE_SYS_QUEUE)
 CHECK_INCLUDE_FILES(sys/un.h HAVE_SYS_UN)
+CHECK_INCLUDE_FILES(onigposix.h HAVE_ONIG_H)
 
 CHECK_TYPE_SIZE("int" SIZEOF_INT)
 CHECK_TYPE_SIZE("long" SIZEOF_LONG)
@@ -88,57 +89,6 @@
 endif()
 	
 
-if (NOT EVHTP_DISABLE_REGEX)
-	CONFIGURE_FILE(${CMAKE_CURRENT_SOURCE_DIR}/oniguruma/config.h.in ${CMAKE_CURRENT_BINARY_DIR}/oniguruma/config.h)
-	set(ONIG_SOURCES
-		oniguruma/regerror.c
-		oniguruma/regparse.c
-		oniguruma/regext.c
-		oniguruma/regcomp.c
-		oniguruma/regexec.c
-		oniguruma/reggnu.c
-		oniguruma/regenc.c
-		oniguruma/regsyntax.c
-		oniguruma/regtrav.c
-		oniguruma/regversion.c
-		oniguruma/st.c
-		oniguruma/regposix.c
-		oniguruma/regposerr.c
-		oniguruma/enc/unicode.c
-		oniguruma/enc/ascii.c
-		oniguruma/enc/utf8.c
-		oniguruma/enc/utf16_be.c
-		oniguruma/enc/utf16_le.c
-		oniguruma/enc/utf32_be.c
-		oniguruma/enc/utf32_le.c
-		oniguruma/enc/euc_jp.c
-		oniguruma/enc/sjis.c
-		oniguruma/enc/iso8859_1.c
-		oniguruma/enc/iso8859_2.c
-		oniguruma/enc/iso8859_3.c
-		oniguruma/enc/iso8859_4.c
-		oniguruma/enc/iso8859_5.c
-		oniguruma/enc/iso8859_6.c
-		oniguruma/enc/iso8859_7.c
-		oniguruma/enc/iso8859_8.c
-		oniguruma/enc/iso8859_9.c
-		oniguruma/enc/iso8859_10.c
-		oniguruma/enc/iso8859_11.c
-		oniguruma/enc/iso8859_13.c
-		oniguruma/enc/iso8859_14.c
-		oniguruma/enc/iso8859_15.c
-		oniguruma/enc/iso8859_16.c
-		oniguruma/enc/euc_tw.c
-		oniguruma/enc/euc_kr.c
-		oniguruma/enc/big5.c
-		oniguruma/enc/gb18030.c
-		oniguruma/enc/koi8_r.c
-		oniguruma/enc/cp1251.c)
-else()
-	set(ONIG_SOURCES "")
-endif()
-
-
 include_directories(
 	${CMAKE_CURRENT_BINARY_DIR}/compat
 	${CMAKE_CURRENT_SOURCE_DIR}/htparse
@@ -171,6 +121,10 @@
 	set (LIBEVHTP_SOURCES ${LIBEVHTP_SOURCES} evthr/evthr.c)
 endif(NOT EVHTP_DISABLE_EVTHR)
 
+if (NOT EVHTP_DISABLE_REGEX)
+	set (LIBEVHTP_EXTERNAL_LIBS ${LIBEVHTP_EXTERNAL_LIBS} onig)
+endif(NOT EVHTP_DISABLE_REGEX)
+
 if (NOT ${CMAKE_BUILD_TYPE} STREQUAL "Debug")
   set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -DNDEBUG")
 endif (NOT ${CMAKE_BUILD_TYPE} STREQUAL "Debug")
@@ -179,14 +133,13 @@
 	set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -DEVHTP_DISABLE_SSL")
 endif()
 
-add_library(libevhtp STATIC ${LIBEVHTP_SOURCES} ${ONIG_SOURCES})
+add_library(libevhtp STATIC ${LIBEVHTP_SOURCES})
 set_target_properties(libevhtp PROPERTIES OUTPUT_NAME "evhtp")
 
 install (TARGETS libevhtp DESTINATION lib)
 install (FILES evhtp.h DESTINATION include)
 install (FILES htparse/htparse.h DESTINATION include)
 install (FILES evthr/evthr.h DESTINATION include)
-install (FILES oniguruma/onigposix.h DESTINATION include)
 
 find_library (LIB_DL dl)
 set (SYS_LIBS ${LIB_DL})
