--- transmission-1.51/configure.ac.orig	2009-03-30 16:54:36.000000000 +0200
+++ transmission-1.51/configure.ac	2009-03-30 16:55:01.000000000 +0200
@@ -21,17 +21,9 @@
 
 if test m4_substr(peer_id_prefix,6,1) = "0"; then
   supported_build=yes
-  if test "x$GCC" = "xyes" ; then
-    CFLAGS="$CFLAGS -g -O3 -funroll-loops "
-    CXXFLAGS="$CXXFLAGS -g -O3 -funroll-loops "
-  fi
   AC_DEFINE(TR_STABLE_RELEASE, 1, [Define to 1 if this is a stable release of Transmission])
 else
   supported_build=no
-  if test "x$GCC" = "xyes" ; then
-    CFLAGS="$CFLAGS -g -O0"
-    CXXFLAGS="$CXXFLAGS -g -O0"
-  fi
   if test m4_substr(peer_id_prefix,6,1) = "X"; then
     AC_DEFINE(TR_BETA_RELEASE, 1, [Define to 1 if this is a beta release of Transmission])
   else
